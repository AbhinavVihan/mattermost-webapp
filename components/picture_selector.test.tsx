// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import PictureSelector from 'components/picture_selector';
import {renderWithIntl} from 'tests/react_testing_utils';

describe('components/picture_selector', () => {
    const baseProps = {
        name: 'picture_selector_test',
        onSelect: jest.fn(),
        onRemove: jest.fn(),
    };

    test('should match snapshot, no picture selected', () => {
        const wrapper = renderWithIntl(
            <PictureSelector {...baseProps}/>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, existing picture provided', () => {
        const props = {
            ...baseProps,
            src: 'http:///url.com/picture.jpg',
            defaultSrc: 'http:///url.com/default-picture.jpg',
        };

        const wrapper = renderWithIntl(
            <PictureSelector {...props}/>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, default picture provided', () => {
        const props = {
            ...baseProps,
            defaultSrc: 'http:///url.com/default-picture.jpg',
        };

        const wrapper = renderWithIntl(
            <PictureSelector {...props}/>,
        );

        expect(wrapper).toMatchSnapshot();
    });
});
