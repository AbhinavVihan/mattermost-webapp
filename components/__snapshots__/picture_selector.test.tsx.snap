// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/picture_selector should match snapshot, default picture provided 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="PictureSelector"
      >
        <input
          accept=".jpeg,.jpg,.png,.bmp"
          aria-hidden="true"
          class="PictureSelector__input hidden"
          data-testid="PictureSelector__input-picture_selector_test"
          name="picture_selector_test"
          tabindex="-1"
          type="file"
        />
        <div
          class="PictureSelector__imageContainer"
        >
          <div
            aria-label="Picture selector image"
            class="PictureSelector__image"
            style="background-image: url(http:///url.com/default-picture.jpg);"
          />
          <button
            aria-label="Select picture"
            class="PictureSelector__selectButton"
            data-testid="PictureSelector__selectButton"
          >
            <i
              class="icon icon-pencil-outline"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="PictureSelector"
    >
      <input
        accept=".jpeg,.jpg,.png,.bmp"
        aria-hidden="true"
        class="PictureSelector__input hidden"
        data-testid="PictureSelector__input-picture_selector_test"
        name="picture_selector_test"
        tabindex="-1"
        type="file"
      />
      <div
        class="PictureSelector__imageContainer"
      >
        <div
          aria-label="Picture selector image"
          class="PictureSelector__image"
          style="background-image: url(http:///url.com/default-picture.jpg);"
        />
        <button
          aria-label="Select picture"
          class="PictureSelector__selectButton"
          data-testid="PictureSelector__selectButton"
        >
          <i
            class="icon icon-pencil-outline"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`components/picture_selector should match snapshot, existing picture provided 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="PictureSelector"
      >
        <input
          accept=".jpeg,.jpg,.png,.bmp"
          aria-hidden="true"
          class="PictureSelector__input hidden"
          data-testid="PictureSelector__input-picture_selector_test"
          name="picture_selector_test"
          tabindex="-1"
          type="file"
        />
        <div
          class="PictureSelector__imageContainer"
        >
          <div
            aria-label="Picture selector image"
            class="PictureSelector__image"
            style="background-image: url(http:///url.com/picture.jpg);"
          />
          <button
            aria-label="Select picture"
            class="PictureSelector__selectButton"
            data-testid="PictureSelector__selectButton"
          >
            <i
              class="icon icon-pencil-outline"
            />
          </button>
        </div>
        <button
          class="PictureSelector__removeButton"
          data-testid="PictureSelector__removeButton"
        >
          Remove picture
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="PictureSelector"
    >
      <input
        accept=".jpeg,.jpg,.png,.bmp"
        aria-hidden="true"
        class="PictureSelector__input hidden"
        data-testid="PictureSelector__input-picture_selector_test"
        name="picture_selector_test"
        tabindex="-1"
        type="file"
      />
      <div
        class="PictureSelector__imageContainer"
      >
        <div
          aria-label="Picture selector image"
          class="PictureSelector__image"
          style="background-image: url(http:///url.com/picture.jpg);"
        />
        <button
          aria-label="Select picture"
          class="PictureSelector__selectButton"
          data-testid="PictureSelector__selectButton"
        >
          <i
            class="icon icon-pencil-outline"
          />
        </button>
      </div>
      <button
        class="PictureSelector__removeButton"
        data-testid="PictureSelector__removeButton"
      >
        Remove picture
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`components/picture_selector should match snapshot, no picture selected 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="PictureSelector"
      >
        <input
          accept=".jpeg,.jpg,.png,.bmp"
          aria-hidden="true"
          class="PictureSelector__input hidden"
          data-testid="PictureSelector__input-picture_selector_test"
          name="picture_selector_test"
          tabindex="-1"
          type="file"
        />
        <div
          class="PictureSelector__imageContainer"
        >
          <div
            aria-label="Picture selector image"
            class="PictureSelector__image"
            style="background-image: url(undefined);"
          />
          <button
            aria-label="Select picture"
            class="PictureSelector__selectButton"
            data-testid="PictureSelector__selectButton"
          >
            <i
              class="icon icon-pencil-outline"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="PictureSelector"
    >
      <input
        accept=".jpeg,.jpg,.png,.bmp"
        aria-hidden="true"
        class="PictureSelector__input hidden"
        data-testid="PictureSelector__input-picture_selector_test"
        name="picture_selector_test"
        tabindex="-1"
        type="file"
      />
      <div
        class="PictureSelector__imageContainer"
      >
        <div
          aria-label="Picture selector image"
          class="PictureSelector__image"
          style="background-image: url(undefined);"
        />
        <button
          aria-label="Select picture"
          class="PictureSelector__selectButton"
          data-testid="PictureSelector__selectButton"
        >
          <i
            class="icon icon-pencil-outline"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
